From: Joe Nahmias <joe@nahmias.net>
Date: Sun, 30 Jun 2013 19:36:17 -0400
Subject: fix crash when copying NULL selections

Description: fix crash when copying NULL selections
Author: Joe Nahmias <joe@nahmias.net>
Bug-Debian: https://bugs.debian.org/596763
Last-Update: 2013-06-30
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
---
 xsel.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/xsel.c b/xsel.c
index 841bd34..4d684b2 100644
--- a/xsel.c
+++ b/xsel.c
@@ -836,8 +836,11 @@ copy_sel (unsigned char * s)
 {
   unsigned char * new_sel = NULL;
 
-  new_sel = xs_strdup (s);
-  current_alloc = total_input = xs_strlen (s);
+  if (s)
+    new_sel = xs_strdup (s);
+  else
+    new_sel = xs_strdup ("");
+  current_alloc = total_input = xs_strlen (new_sel);
 
   return new_sel;
 }
